gitpod-io/gitpod
# jetbrains: launch with interactive login shell in async manner

## Description


This PR ensures:
- JB GW can try to connect to JB backend as soon as workspace content and .dotflies are initialised, before we will wait for all preparation steps to run the backend before letting a user to attempt
- JB backend is started with proper user envs, i.e. collected from the login user shell, before they were ignored

After this PR is merged we should try to revert work arounds for detecting JDK and retest. We should also see whether other products can detect properly other SDKs, i.e. golang for gitpod repo, rubymine ruby, pycharm python and so on. cc @felladrin @andreafalzetti 

## Related Issue(s)

fix #13815

## How to test


- Start a workspace with JB product
- Export some env var in `/etc/profile`
- Run `kill <pid-of-status>` to emulate graceful shutdown
- When JB backend is restarted like that, investigate env vars of the backend `/proc/<pid-of-backend>/environ`. It should have env var from `/etc/profile`
- Investigate logs of workplace you should see `asked IDE to terminate` meaning that JB backend was restarted gracefully via codeWithMe endpoint [[1](https://github.com/gitpod-io/gitpod/blob/8b1eff9e06a8b37ea12ef4b730fb690ea7350797/components/ide/jetbrains/image/status/main.go#L302)]

## Release Notes

```release-note
JetBrains: Start JB backend with the interactive login shell.
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
